core/src/unprocessed_packet_batches.rs

```diff
@@ -34,6 +33,52 @@ pub struct DeserializedPacketBatch {
     pub unprocessed_packets: HashMap<usize, DeserializedPacket>,
 }
 
+pub struct UnprocessedPacketBatches(VecDeque<DeserializedPacketBatch>);
```

Comment:
Can you derive Debug and Default here? Debug is helpful for future-us, and Default can replace having to manually implement it below (then `new()` can call `Self::default()`).

Reply:
Derived `Default`, but don't think want to derive `Debug`. The buffer could be large - thousands of entries. When need to peek into details, its item `DeserializedPacketBatch` has `Debug` trait, that can be printed.